pkgs/applications/editors/emacs/generic.nix

```diff
@@ -67,7 +67,35 @@ let emacs = (if withMacport then llvmPackages_6.stdenv else stdenv).mkDerivation
   pname = pname + lib.optionalString ( !withX && !withNS && !withMacport && !withGTK2 && !withGTK3 ) "-nox";
   inherit version;
 
-  patches = patches fetchpatch;
+  patches = patches fetchpatch ++
+            lib.optionals nativeComp
+              [
+                (substituteAll {
+                  src = let majorVersionStr = lib.versions.major version;
+                        in
+                          # The second part of the OR branch accounts
+                          # for people who use a pinned rev of the
+                          # Emacs overlay.  Once Emacs 29 is stable,
+                          # those people will probably switch to the
+                          # release, so we can remove that branch at
+                          # that time.
+                          if lib.versionOlder version "29" || (lib.stringLength majorVersionStr == 8 && lib.versionOlder version "20221003")
+                          then ./native-comp-driver-options-28.patch
+                          else ./native-comp-driver-options.patch;
```

Comment:
Emacs 29 is released. I doubt anyone is still using >1 month old development branch.